src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/RawBinaryReader.cpp

```diff
@@ -0,0 +1,245 @@
+
+Result<> RawBinaryReader::operator()()
+{
+  return execute();
+}
+
+// -----------------------------------------------------------------------------
+Result<> RawBinaryReader::execute()
+{
+  IDataArray* binaryIDataArray = m_DataStructure.getDataAs<IDataArray>(m_InputValues->createdAttributeArrayPathValue);
+  if(binaryIDataArray == nullptr)
+  {
+    return MakeErrorResult(k_RbrDAError, "Can't obtain DataArray from path '" + m_InputValues->createdAttributeArrayPathValue.toString() + "'.");
+  }
+
+  if(binaryIDataArray->getNumberOfComponents() != static_cast<size_t>(m_InputValues->numberOfComponentsValue))
+  {
+    return MakeErrorResult(k_RbrComponentError, "Failed to acquire DataArray from path '" + m_InputValues->createdAttributeArrayPathValue.toString() + "' with the correct number of components.");
+  }
+
+  const std::string inputFile = m_InputValues->inputFileValue.string();
+
+  int32_t err = 0;
+  switch(m_InputValues->scalarTypeValue)
+  {
+  case NumericType::int8:
+    err = readBinaryFile<int8_t>(binaryIDataArray, inputFile, m_InputValues->skipHeaderBytesValue, m_InputValues->endianValue);
+    break;
+  case NumericType::uint8:
+    err = readBinaryFile<uint8_t>(binaryIDataArray, inputFile, m_InputValues->skipHeaderBytesValue, m_InputValues->endianValue);
+    break;
+  case NumericType::int16:
+    err = readBinaryFile<int16_t>(binaryIDataArray, inputFile, m_InputValues->skipHeaderBytesValue, m_InputValues->endianValue);
+    break;
+  case NumericType::uint16:
+    err = readBinaryFile<uint16_t>(binaryIDataArray, inputFile, m_InputValues->skipHeaderBytesValue, m_InputValues->endianValue);
+    break;
+  case NumericType::int32:
+    err = readBinaryFile<int32_t>(binaryIDataArray, inputFile, m_InputValues->skipHeaderBytesValue, m_InputValues->endianValue);
+    break;
+  case NumericType::uint32:
+    err = readBinaryFile<uint32_t>(binaryIDataArray, inputFile, m_InputValues->skipHeaderBytesValue, m_InputValues->endianValue);
+    break;
+  case NumericType::int64:
+    err = readBinaryFile<int64_t>(binaryIDataArray, inputFile, m_InputValues->skipHeaderBytesValue, m_InputValues->endianValue);
+    break;
+  case NumericType::uint64:
+    err = readBinaryFile<uint64_t>(binaryIDataArray, inputFile, m_InputValues->skipHeaderBytesValue, m_InputValues->endianValue);
+    break;
+  case NumericType::float32:
+    err = readBinaryFile<float>(binaryIDataArray, inputFile, m_InputValues->skipHeaderBytesValue, m_InputValues->endianValue);
+    break;
+  case NumericType::float64:
+    err = readBinaryFile<double>(binaryIDataArray, inputFile, m_InputValues->skipHeaderBytesValue, m_InputValues->endianValue);
+    break;
+    //  case NumericType::boolean:
+    //    err = readBinaryFile<uint8_t>(binaryIDataArray, inputFile, m_InputValues->skipHeaderBytesValue, m_InputValues->endianValue);
+    //    break;
+  default:
+    throw std::runtime_error("The chosen scalar type is not supported.");
```

Comment:
Don't throw a runtime error if we can return an error code. The calling code would need to check the error codes anyways so just return an "MakeErrorResult()"